src/mscorlib/src/System/Runtime/CompilerServices/ReadOnlyAttribute.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System;
+
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.All, Inherited = false)]
```

Comment:
> I think the entire CompilerServices namespace is already meant for compiler consumption (but not general consumption), so may be superfluous

There are types in the compiler namespace that folks use, though, like ConditionalWeakTable, StrongBox, even AsyncTaskMethodBuilder.  I think adding the EditorBrowsable attribute would be reasonable.